scripts/ci/prepare_tool_scripts.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -euo pipefail
+
+function prepare_tool_script() {
+    IMAGE="${1}"
+    VOLUME="${2}"
+    TOOL="${3}"
+    COMMAND="${4:-}"
+
+    TARGET_TOOL_PATH="/usr/bin/${TOOL}"
+    TARGET_TOOL_UPDATE_PATH="/usr/bin/${TOOL}-update"
+
+    cat >"${TARGET_TOOL_PATH}" <<EOF
+#!/usr/bin/env bash
```

Comment:
What do you think about creating one generic script and many symbolic links? This one script can select an image based on "basename "${BASH_SOURCE[0]}".  If it detects suffix "-update" it will update the image.  The script can be located in one file e.g. scripts/ci/in_container/run_cli_tools.sh.

Reply:
Busbox works in a similar way. 

> 
> BusyBox is a multi-call binary. A multi-call binary is an executable program that performs the same job as more than one utility program. That means there is just a single BusyBox binary, but that single binary acts like a large number of utilities. This allows BusyBox to be smaller since all the built-in utility programs (we call them applets) can share code for many common operations.
> 
> You can also invoke BusyBox by issuing a command as an argument on the command line. For example, entering
> 
>         /bin/busybox ls
> will also cause BusyBox to behave as 'ls'.
> 
> Of course, adding '/bin/busybox' into every command would be painful. So most people will invoke BusyBox using links to the BusyBox binary.
> 
> For example, entering
> 
>         ln -s /bin/busybox ls
>         ./ls
> will cause BusyBox to behave as 'ls' (if the 'ls' command has been compiled into BusyBox). Generally speaking, you should never need to make all these links yourself, as the BusyBox build system will do this for you when you run the 'make install' command.
> 
> If you invoke BusyBox with no arguments, it will provide you with a list of the applets that have been compiled into your BusyBox binary.

https://busybox.net/downloads/BusyBox.html

Reply:
It will not work for 'aws' for example - because the command is a bit different there. I think I prefer to generate separate scripts - also because in the future we might add other tools like that and they might have different commands to run. having separate script. I prefer to have different logic to generate those scripts.

For busybox it makes perfect sense because it is quite substantial binary and copying it over takes precious space. In our case it's not an issue and I prefer each tool is separate